test/beast/http/file_body.cpp

```diff
@@ -98,9 +136,89 @@ class file_body_test : public beast::unit_test::suite
         boost::filesystem::remove(temp, ec);
         BEAST_EXPECTS(! ec, ec.message());
     }
+
+#if BOOST_BEAST_USE_POSIX_FILE
+    static void move_to_eof(file_posix& file)
+    {
+        ::lseek(file.native_handle(), 0, SEEK_END);
+    }
+#endif
+
+    static void move_to_eof(file_stdio& file)
+    {
+        std::fseek(file.native_handle(), 0, SEEK_END);
+    }
+
+#if BOOST_BEAST_USE_WIN32_FILE
+    static void move_to_eof(file_win32& file)
+    {
+        ::SetFilePointer(file.native_handle(), 0, 0, FILE_END);
+    }
+#endif
+
+    template<class FileType>
+    void
+    fileBodyUnexpectedEofOnGet()
+    {
+        auto temp = temp_file(log);
+
+        error_code ec;
+        string_view const ten =
+            "0123456789"; // 40
+        // create the temporary file
+        {
+            std::ofstream fstemp(temp.path().native());
+            std::size_t written = 0;
+            std::size_t to_write = 4097;
+            while (written < to_write)
+            {
+                auto size = std::min(ten.size(), to_write - written);
+                fstemp << ten.substr(0, size);
+                BEAST_EXPECT(fstemp.good());
+                written += size;
+            }
+            fstemp.close();
+        }
+
+        // open the file and read the header
+        {
+            using file_body_type = basic_file_body<FileType>;
+
+            typename file_body_type::value_type value;
+            // opened in write mode so we can truncate later
+            value.open(temp.path().native().c_str(), file_mode::read, ec);
+            BEAST_EXPECTS(! ec, ec.message());
+
+            response_header<> header;
+            header.version(11);
+            header.result(status::accepted);
+            header.set(field::server, "test");
+            header.set(field::content_length, 4097);
+
+            typename file_body_type::writer w(header, value);
+            auto maybe_range = w.get(ec);
+            BEAST_EXPECTS(!ec, ec.message());
+            BEAST_EXPECTS(maybe_range.has_value(), "no range returned");
+            BEAST_EXPECT(maybe_range.value().second);
+
+            move_to_eof(value.file());
+            maybe_range = w.get(ec);
+            BEAST_EXPECTS(ec == net::error::eof, ec.message());
+            BEAST_EXPECTS(!maybe_range.has_value(), "range returned on error");
+        }
+    }
+
     void
     run() override
     {
+        fileBodyUnexpectedEofOnGet<file_stdio>();
+    #if BOOST_BEAST_USE_POSIX_FILE
+        fileBodyUnexpectedEofOnGet<file_posix>();
+    #endif
+    #if BOOST_BEAST_USE_WIN32_FILE
+        fileBodyUnexpectedEofOnGet<file_win32>();
+    #endif
```

Comment:
Please move these calls to below the calls to `doTestFileBody` so that the order of invocation matches the order of definitions in the source file.